romancal/skycell/skymap.py

```diff
@@ -0,0 +1,588 @@
+        log.info(
+            f"Creating skycell image at ra: {skycell.radec_center[0]}  dec {skycell.radec_center[1]}",
+        )
+
+        return skycell.wcs
+    except ValueError:
+        return None
+
+
+class SkyMap:
+    """
+    There are 8 million sky cells in the table, ~2000 per skytile.
+    """
+
+    __path: None | Path
+
+    def __init__(self, path: None | Path | str = None):
+        if path is not None and not isinstance(path, Path):
+            path = Path(path)
+        self.__path = path
+
+    @property
+    def path(self) -> None | Path:
+        return self.__path
+
+    @cached_property
+    def data(self) -> AsdfFile:
+        if self.__path is None:
+            rmap = crds.getreferences(
+                {
+                    "roman.meta.instrument.name": "WFI",
+                    "roman.meta.exposure.start_time": "2000-01-01 00:00:00",
+                },
+                reftypes=["skycells"],
+                context="roman_0081.pmap",
+                observatory="roman",
+            )
+            self.__path = Path(rmap["skycells"])
+        with asdf.open(self.__path, memmap=True) as file:
+            output = file.copy()
+        return output
+
+    @property
+    def skycells(self) -> np.void:
+        return self.data["roman"]["skycells"]
+
+    @cached_property
+    def skycell_kdtree(self) -> KDTree:
+        """
+        k-d tree of normalized center vectorpoints in 3D space of each skycell
+        """
+
+        return KDTree(
+            sgv.normalize_vector(
+                np.stack(
+                    sgv.lonlat_to_vector(
+                        self.skycells["ra_center"], self.skycells["dec_center"]
+                    ),
+                    axis=1,
+                )
+            )
+        )
+
+    @property
+    def skytiles(self) -> np.void:
+        """
+        skytiles (projection regions)
+        """
+
+        return self.data["roman"]["projection_regions"]
+
+    @cached_property
+    def skytile_kdtree(self) -> KDTree:
+        """
+        k-d tree of normalized center vectorpoints in 3D space of each skytile
+        """
+
+        return KDTree(
+            sgv.normalize_vector(
+                np.stack(
+                    sgv.lonlat_to_vector(
+                        self.skytiles["ra_tangent"], self.skytiles["dec_tangent"]
+                    ),
+                    axis=1,
+                )
+            )
+        )
+
+    @property
+    def pixel_scale(self) -> float:
+        # The scale is given in arcseconds per pixel. Convert to degrees.
+        return self.data["roman"]["meta"]["pixel_scale"] / 3600.0
+
+    @property
+    def pixel_shape(self) -> tuple[int, int]:
+        return self.data["roman"]["meta"]["nxy_skycell"], self.data["roman"]["meta"][
+            "nxy_skycell"
+        ]
+
+    def __getitem__(self, index: int) -> SkyCell:
+        return SkyCell(self.data["roman"]["skycells"][index])
```

Comment:
Is this supposed to be `SkyCell.from_data(self.data["roman"]["skycells"][index])`?